javaparser-symbol-solver-testing/src/test/java/com/github/javaparser/symbolsolver/Issue1668.java

```diff
@@ -0,0 +1,62 @@
+package com.github.javaparser.symbolsolver;
+
+import com.github.javaparser.*;
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.body.VariableDeclarator;
+import com.github.javaparser.ast.expr.VariableDeclarationExpr;
+import com.github.javaparser.resolution.declarations.ResolvedFieldDeclaration;
+import com.github.javaparser.resolution.types.ResolvedType;
+import com.github.javaparser.symbolsolver.model.resolution.TypeSolver;
+import com.github.javaparser.symbolsolver.resolution.typesolvers.ReflectionTypeSolver;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class Issue1668 {
+
+    private JavaParser javaParser;
+
+    @Before
+    public void setUp() {
+        TypeSolver typeSolver = new ReflectionTypeSolver();
+        ParserConfiguration config = new ParserConfiguration();
+        config.setSymbolResolver(new JavaSymbolSolver(typeSolver));
+        javaParser = new JavaParser(config);
+    }
+
+    @Test
+    public void test() {
+        String code = String.join(System.lineSeparator(),
+                "public class X {",
+                "   public static void main(String[] args) {",
+                "       String s = \"a,b,c,d,e\";",
+                "       String[] stringArray = s.split(',');",
+                "   }",
+                "}"
+        );
+
+        ParseResult<CompilationUnit> parseResult = javaParser.parse(ParseStart.COMPILATION_UNIT, Providers.provider(code));
+        assertTrue(parseResult.isSuccessful());
+        assertTrue(parseResult.getResult().isPresent());
+
+        CompilationUnit compilationUnit = parseResult.getResult().get();
+        Optional<VariableDeclarator> stringArray = compilationUnit.findFirst(VariableDeclarator.class, variableDeclarator ->
+                variableDeclarator.getNameAsString()
+                        .equals("stringArray"));
+        assertTrue(stringArray.isPresent());
+        VariableDeclarator variableDeclarator = stringArray.get();
+        variableDeclarator.getParentNode().ifPresent(parent -> {
+            if (parent instanceof VariableDeclarationExpr) {
+                VariableDeclarationExpr variableDeclarationExpr = (VariableDeclarationExpr) parent;
+                ResolvedType resolvedType = variableDeclarationExpr.calculateResolvedType();
+                assertEquals("java.lang.String[]", resolvedType.describe());
+            }
+        });
+        ResolvedFieldDeclaration resolve = variableDeclarator.resolve();
+        assertEquals("java.lang.String[]", resolve.declaringType().getQualifiedName());
```

Comment:
mmm... can the qualified name corresponds to an array type?

Reply:
A qualified name is identifier.identifier.identifier etc. How should an array type be represented?

Reply:
Anyway the declaring type of a field cannot be an array type, it must be a class, an interface or an enum

Reply:
So java.lang.String in this case?

Reply:
In this case it does not make much sense becose `stringArray` is not a field... so I would not expect this to work.

Reply:
But in this case:
```
class A {
   int foo;
}
```

The declaring type of `foo` is `A`.

Reply:
The reason I used an array type as qualified name is because that is what is returned by `variableDeclarationExpr.calculateResolvedType().describe`. I don't know what it should return, but I had to fill in something.

Reply:
@ftomassetti @daanschipper can you come to a resolution on this?

Reply:
I would expect this to fail. If it does not let leave it as it is

Reply:
Double check: you would expect this test to fail, so the whole testcase and the connected issue are to be closed?